src/control/positionControl/PositionControl.cpp

```diff
@@ -6,44 +6,215 @@
+        }
+        if (shouldRecalculatePath(currentPosition, targetPosition, currentVelocity, robotId)) {
+            computedPaths[robotId] = pathPlanningAlgorithm.computePath(currentPosition, targetPosition);
+        }
+        interface::Input::drawData(interface::Visual::PATHFINDING, computedPaths[robotId], Qt::green, robotId, interface::Drawing::LINES_CONNECTED);
+        interface::Input::drawData(interface::Visual::PATHFINDING, {computedPaths[robotId].front(), currentPosition}, Qt::green, robotId, interface::Drawing::LINES_CONNECTED);
+        interface::Input::drawData(interface::Visual::PATHFINDING, computedPaths[robotId], Qt::blue, robotId, interface::Drawing::DOTS);
+
+
+        RobotCommand command = RobotCommand();
+        command.pos = computedPaths[robotId].front();
+        Position trackingVelocity = pathTrackingAlgorithm.trackPathDefaultAngle(currentPosition, currentVelocity,computedPaths[robotId], robotId,pidType);
+        command.vel = Vector2(trackingVelocity.x, trackingVelocity.y);
+        command.angle = trackingVelocity.rot;
+        return command;
     }
 
-    // if the robot is close to the final position and can't get there, stop
-    if ((currentPosition - targetPosition).length() < FINAL_AVOIDANCE_DISTANCE && collisionDetector.getRobotCollisionBetweenPoints(currentPosition, targetPosition)) {
-        RTT_INFO("Path collides with something close to the target position for robot ID ", robotId)
-        return {};
+    bool PositionControl::shouldRecalculatePath(const Vector2 &currentPosition, const Vector2 &targetPos,
+                                                const Vector2 &currentVelocity, int robotId) {
+        return computedPaths[robotId].empty() ||
+               PositionControlUtils::isTargetChanged(targetPos, computedPaths[robotId].back()) ||
+               (currentVelocity != Vector2(0, 0) &&
+                collisionDetector.isCollisionBetweenPoints(currentPosition, computedPaths[robotId].front()));
+    }
+
+    void PositionControl::setRobotPositions(std::vector<Vector2> &robotPositions) {
+        collisionDetector.setRobotPositions(robotPositions);
+    }
+
+std::pair<RobotCommand, std::optional<Vector2>> PositionControl::computeAndTrackPathBBT(const rtt::world::Field &field, int robotId, Vector2 currentPosition, Vector2 currentVelocity,
+                                 Vector2 targetPosition, stp::PIDType pidType) {
+        //TODO: Create a function in WorldObjects which can give the position of the first collision to STP
+        //TODO: find a good value for the timeStep
+        double timeStep = 0.1;
+
+        worldObjects.setField(field);
+        std::optional<BB::CollisionData> firstCollision;
+        std::pair<RobotCommand, std::optional<Vector2>> robotCommandCollisionPair;
+        // Currently calculate all paths again on each tick because the way the path is used in control is not made for the BBT
+        if (true || (!computedPathsBB.contains(robotId) ||
```

Comment:
the true is suspicious? Is it related to the comment above?

Reply:
Yes, I did not get it working yet. Right now it only works when recalculating the path every time but I wanted to have the logic in place when that is fixed. So the true is only there for now to calculate the path every tick.

Reply:
I added an extra comment to clarify it